src/core/batch.rkt

```diff
@@ -142,6 +147,10 @@
     (vector-ref mappings idx))
 
   (define out (make-mutable-batch))
+  (when keep-vars
+    (for ([var (in-list (batch-vars input-batch))])
+      (batch-push! out var)))
```

Comment:
This is an edge case and triggered by compiler.rkt
The scenario is:

`make-compiler` receives an empty expression `'()` with some list of variables `'(x y z)`.
When creating a batch out of these, the nodes look like: `#(x y z)`, the rootvec is empty `#()`.
We remove `spec` from this batch by using `batch-remove-approx` which is a part of our `make-compiler` pipeline.
And after removing `spec` we potentially may have zombie nodes in the batch.
Therefore, we trigger `batch-remove-zombie` but since the original expression didn't have any roots at all - the variables are going to be removed.
We do not want that and actually these variables in this edge case should be preserved. 
`keep-vars` flag takes care of this edge case